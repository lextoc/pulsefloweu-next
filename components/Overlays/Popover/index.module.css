.menuButton {
  cursor: pointer;
  border: 0;
  background: 0;
  width: 2rem;
  height: 2rem;
  border-radius: 2rem;
  display: flex;
  justify-content: center;
  align-items: center;
  transform: translateY(0.05rem);
}

.menuButton:focus,
.menuButton:hover {
  background-color: rgb(var(--light-background));
}

.menuButtonWhite:focus,
.menuButtonWhite:hover {
  background-color: rgba(var(--light-background), 0.1);
}

.menu {
  background: white;
  border-radius: var(--border-radius);
  box-shadow: 0 0.5rem 1rem rgba(var(--secondary), 0.1);
  overflow: hidden;
}

.dots {
  font-size: 1rem;
  height: 1rem;
}

.blankButton {
  font: inherit;
  cursor: pointer;
  width: 100%;
}
