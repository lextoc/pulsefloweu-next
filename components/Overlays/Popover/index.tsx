import { IconDotsVertical } from "@tabler/icons-react";
import { useState } from "react";
import { Popover } from "react-tiny-popover";

import styles from "./index.module.css";

export interface PopoverProps {
  content: React.ReactNode;
  white?: boolean;
}

export default function ({ content, white }: PopoverProps) {
  const [isOpen, setIsOpen] = useState<boolean>(false);
  return (
    <Popover
      isOpen={isOpen}
      positions={["bottom"]}
      onClickOutside={() => setIsOpen(false)}
      containerStyle={{ zIndex: "250" }}
      content={
        <div className={styles.menu} onClick={() => setIsOpen(false)}>
          {content}
        </div>
      }
    >
      <button
        className={`${styles.menuButton} ${
          white ? styles.menuButtonWhite : ""
        }`}
        onClick={() => setIsOpen(!isOpen)}
      >
        <IconDotsVertical
          className={styles.dots}
          color={white ? "white" : "rgb(6, 31, 54)"}
        />
      </button>
    </Popover>
  );
}
