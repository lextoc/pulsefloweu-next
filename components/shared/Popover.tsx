import { IconDotsVertical } from "@tabler/icons-react";
import { useState } from "react";
import { Popover } from "react-tiny-popover";

import styles from "./Popover.module.css";

export interface IPopoverProps {
  content: React.ReactNode;
  white?: boolean;
}

export default function ({ content, white }: IPopoverProps) {
  const [isOpen, setIsOpen] = useState<boolean>(false);
  return (
    <Popover
      isOpen={isOpen}
      positions={["bottom"]}
      onClickOutside={() => setIsOpen(false)}
      containerStyle={{ zIndex: "250" }}
      content={
        <div className={styles.menu} onClick={() => setIsOpen(false)}>
          {content}
        </div>
      }
    >
      <button
        className={`${styles.menuButton} ${
          white ? styles.menuButtonWhite : ""
        }`}
        onClick={() => setIsOpen(!isOpen)}
      >
        <IconDotsVertical
          size="1rem"
          color={white ? "white" : "rgb(6, 31, 54)"}
        />
      </button>
    </Popover>
  );
}
