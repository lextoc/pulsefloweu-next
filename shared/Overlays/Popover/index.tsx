import { IconDotsVertical } from "@tabler/icons-react";
import { useState } from "react";
import { Popover } from "react-tiny-popover";

import styles from "./index.module.css";

export interface PopoverProps {
  content: React.ReactNode;
  white?: boolean;
  button?: React.ReactNode;
  id?: string;
}

export default function ({ content, white, button, id }: PopoverProps) {
  const [isOpen, setIsOpen] = useState<boolean>(false);
  return (
    <Popover
      isOpen={isOpen}
      positions={["bottom"]}
      onClickOutside={() => setIsOpen(false)}
      containerStyle={{ zIndex: "2000" }}
      content={
        <div className={styles.menu} onClick={() => setIsOpen(false)}>
          {content}
        </div>
      }
    >
      {button ? (
        <button
          id={id}
          className={styles.blankButton}
          onClick={() => setIsOpen(!isOpen)}
        >
          {button}
        </button>
      ) : (
        <button
          id={id}
          className={`${styles.menuButton} ${
            white ? styles.menuButtonWhite : ""
          }`}
          onClick={() => setIsOpen(!isOpen)}
        >
          <IconDotsVertical
            className={styles.dots}
            color={white ? "white" : "rgb(6, 31, 54)"}
            size="1rem"
          />
        </button>
      )}
    </Popover>
  );
}
