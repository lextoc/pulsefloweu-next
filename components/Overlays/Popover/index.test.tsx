import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";

import { PopoverProps } from "./index";
import Popover from "./index";

describe("Popover component", () => {
  const defaultContent = <div>Popover Content</div>;
  const defaultButton = <div>Click me</div>;
  const defaultProps: PopoverProps = {
    content: defaultContent,
    button: defaultButton,
  };

  it("renders the popover button and content", () => {
    render(<Popover {...defaultProps} />);

    const popoverButton = screen.getByRole("button", { name: "Click me" });
    expect(popoverButton).toBeInTheDocument();

    fireEvent.click(popoverButton);

    const popoverContent = screen.getByText("Popover Content");
    expect(popoverContent).toBeInTheDocument();
  });

  it("toggles popover content when button is clicked", () => {
    render(<Popover {...defaultProps} />);

    const popoverButton = screen.getByRole("button", { name: "Click me" });
    expect(popoverButton).toBeInTheDocument();

    fireEvent.click(popoverButton);

    const popoverContent = screen.getByText("Popover Content");
    expect(popoverContent).toBeInTheDocument();

    fireEvent.click(popoverButton);

    expect(popoverContent).not.toBeInTheDocument();
  });

  it("renders custom button and content", () => {
    const customButton = <div>Custom Button</div>;
    const customContent = <div>Custom Content</div>;
    const customProps: PopoverProps = {
      content: customContent,
      button: customButton,
    };

    render(<Popover {...customProps} />);

    const popoverButton = screen.getByRole("button", {
      name: /Custom Button/i,
    });
    expect(popoverButton).toBeInTheDocument();

    fireEvent.click(popoverButton);

    const popoverContent = screen.getByText("Custom Content");
    expect(popoverContent).toBeInTheDocument();
  });
});
