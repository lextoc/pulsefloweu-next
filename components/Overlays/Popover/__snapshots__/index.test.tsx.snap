// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover component matches snapshot 1`] = `
<div>
  <button
    class="blankButton"
  >
    <div>
      Click me
    </div>
  </button>
</div>
`;
